src/test/kotlin/com/esosa/f5pi_backend/services/implementations/UserServiceTest.kt

```diff
@@ -0,0 +1,333 @@
+            )
+        )
+
+        val fieldsPage = PageImpl(testFields)
+
+        `when`(
+            fieldService.getUserFields(
+                result1,
+                0,
+                10,
+                "name",
+                "asc"
+            )
+        ).thenReturn(fieldsPage)
+
+        val result2 = userService.getUserFields(
+            user.id,
+            0,
+            10,
+            "name",
+            "asc"
+        )
+
+        assertEquals(2, result2.numberOfElements)
+    }
+
+    @Test
+    fun getUserSeasons() {
+        val user = User(
+            username = "testUser",
+            password = "testPassword",
+            fullName = "testFullName",
+            email = "test@email.com",
+            role = Role.USER,
+        )
+        `when`(userRepository.save(user)).thenReturn(user)
+
+        val result1 = userService.saveUser(user)
+
+        `when`(userRepository.findById(result1.id)).thenReturn(Optional.of(result1))
+
+        val testSeasons = listOf(
+            SeasonResponse(
+                id = UUID.randomUUID(),
+                name = "2025",
+                initialDate = LocalDate.now().minusDays(30),
+                finalDate = LocalDate.now()
+            ),
+            SeasonResponse(
+                id = UUID.randomUUID(),
+                name = "2024",
+                initialDate = LocalDate.now().minusDays(30),
+                finalDate = LocalDate.now()
+            )
+        )
+
+        val seasonsPage = PageImpl(testSeasons)
+
+        `when`(
+            seasonService.getUserSeasons(
+                result1,
+                0,
+                10,
+                "name",
+                "asc"
+            )
+        ).thenReturn(seasonsPage)
+
+        val result2 = userService.getUserSeasons(
+            user.id,
+            0,
+            10,
+            "name",
+            "asc"
+        )
+
+        assertEquals(2, result2.numberOfElements)
+    }
+
+    @Test
+    fun updateUser() {
+        val user = User(
+            username = "testUser",
+            password = "testPassword",
+            fullName = "testFullName",
+            email = "test@email.com",
+            role = Role.USER
+        )
+        `when`(userRepository.save(user)).thenReturn(user)
+
+        val result1 = userService.saveUser(user)
+
+        val updateUserRequest = UpdateUserRequest(
+            fullName = "testFullName2",
+            email = "test2@email.com",
+        )
+        val modifiedUser = User("testUser", "testPassword", "testFullName2", "test2@email.com", user.role)
+
+        `when`(userRepository.findById(result1.id)).thenReturn(Optional.of(result1))
+        `when`(userRepository.save(result1)).thenReturn(modifiedUser)
+
+        val result2 = userService.updateUser(
+            user.id,
+            updateUserRequest
+        )
+        assertEquals(modifiedUser.fullName, result2.fullName)
+        assertEquals(modifiedUser.email, result2.email)
```

Comment:
Usemos el soporte funcional nativo de Kotlin

```suggestion
        userService.updateUser(TEST_USER.id, updateUserRequest)
            .also { 
                assertEquals(modifiedUser.fullName, it.fullName)
                assertEquals(modifiedUser.email, it.email) 
            }
```